app/lib/view/core/information_display/MealPreviewImage.dart

```diff
@@ -0,0 +1,106 @@
+import 'package:app/view/core/buttons/MensaButton.dart';
+import 'package:app/view_model/repository/data_classes/meal/Meal.dart';
+import 'package:flutter/material.dart';
+
+/// Displays a Meal's image.
+class MealPreviewImage extends StatelessWidget {
+  final Meal _meal;
+  final bool _enableUploadButton;
+  final bool _enableFavoriteButton;
+  final BorderRadius _borderRadius;
+  final double? _height;
+  final double? _width;
+
+  /// Creates a MealPreviewImage.
+  /// @param meal The Meal to display.
+  /// @param enableUploadButton Whether to display the upload button if no image exists.
+  /// @param enableFavoriteButton Whether to display the favorite button.
+  /// @param borderRadius The border radius of the image.
+  /// @param height The height of the image.
+  /// @param width The width of the image.
+  /// @param key The key to use for this widget.
+  /// @return A MealPreviewImage.
+  const MealPreviewImage(
+      {super.key,
+      required Meal meal,
+      bool displayFavorite = false,
+      bool enableUploadButton = false,
+      bool enableFavoriteButton = false,
+      BorderRadius? borderRadius,
+      double? height,
+      double? width})
+      : _meal = meal,
+        _enableUploadButton = enableUploadButton,
+        _enableFavoriteButton = enableFavoriteButton,
+        _borderRadius = borderRadius ?? const BorderRadius.all(Radius.zero),
+        _height = height,
+        _width = width;
+
+  @override
+  Widget build(BuildContext context) {
+    if (_meal.images == null || _meal.images!.isEmpty) {
+      return Container(
+          width: _width,
+          height: _height,
+          decoration: BoxDecoration(
+            borderRadius: _borderRadius,
+            color: Theme.of(context).colorScheme.surface,
+            /*image: const DecorationImage(
+              image: AssetImage('assets/images/meal_placeholder.jpg'),
+              fit: BoxFit.cover,
+            ),*/
+          ),
+          child: ClipRRect(
+              borderRadius: _borderRadius,
+              child: Stack(children: [
+                Center(
+                    child: Column(
+                  mainAxisAlignment: MainAxisAlignment.center,
+                  children: [
+                    Icon(Icons.image_not_supported_outlined,
+                        size: 32,
+                        color: Theme.of(context).colorScheme.onSurface),
+                    if (_enableUploadButton) const SizedBox(height: 16),
+                    if (_enableUploadButton)
+                      MensaButton(onPressed: () {}, text: "Bild hochladen")
```

Comment:
Haven't you already written your interfaces so that you can call these actions?
So you need to come back here later to fill out these actions?
At least that's how we have done things in the backend.